akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/RaffleExample.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package jdocs.akka.persistence.typed;
+
+import akka.actor.typed.ActorSystem;
+import akka.actor.typed.Behavior;
+import akka.persistence.typed.PersistenceId;
+import akka.persistence.typed.javadsl.CommandHandler;
+import akka.persistence.typed.javadsl.CommandHandlerBuilder;
+import akka.persistence.typed.javadsl.Effect;
+import akka.persistence.typed.javadsl.EventHandler;
+import akka.persistence.typed.javadsl.PersistentBehavior;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Random;
+
+/**
+ * Based on
+ * https://github.com/fun-cqrs/fun-cqrs/blob/develop/samples/raffle/src/main/scala/raffle/domain/model/Raffle.scala
+ *
+ * but in Java with a mutable domain object (state) that is independent of Akka
+ */
+public class RaffleExample {
+
+  public static class RaffleId {}
+
+  interface Event {}
+  public static final class RaffleStarted implements Event { }
+  public static final class ParticipantAdded implements Event {
+    public final String name;
+    public ParticipantAdded(String name) {
+      this.name = name;
+    }
+  }
+  public static final class ParticipantRemoved implements Event {
+    public final String name;
+    public ParticipantRemoved(String name) {
+      this.name = name;
+    }
+  }
+  public static final class RaffleCompleted implements Event {}
+
+  interface Command {}
+  public static final class StartRaffle implements Command { }
+
+  public static final class AddParticipant implements Command {
+    public final String name;
+    public AddParticipant(String name) {
+      this.name = name;
+    }
+  }
+
+  public static final class RemoveParticipant implements Command {
+    public final String name;
+    public RemoveParticipant(String name) {
+      this.name = name;
+    }
+  }
+  public static final class CompleteRaffle implements Command { }
+
+  // TODO replies are not implemented in this example yet
+
+  private enum RaffleState {
+    NOT_STARTED,
+    RUNNING,
+    COMPLETED
+  }
+
+  // mutable domain model impl, completely free from Akka stuff
```

Comment:
I like this approach. Both the separation of the domain model and the Akka specific stuff, and the mutable approach, which feels more idiomatic Java.